backend/Testing/Browser/Base/PageTest.cs

```diff
@@ -26,6 +31,23 @@ public PageTest()
     public ILocatorAssertions Expect(ILocator locator) => Assertions.Expect(locator);
     public IPageAssertions Expect(IPage page) => Assertions.Expect(page);
     public IAPIResponseAssertions Expect(IAPIResponse response) => Assertions.Expect(response);
+    /// <summary>
+    /// Consumes a deferred exception that was "thrown" in a sub-thread, and returns it
+    /// or throws if no exception of the given type is found.
+    /// </summary>
+    public UnexpectedResponseException ExpectDeferredException()
+    {
+        DeferredExceptions.Count.ShouldBeGreaterThan(0, "Expected a deferred exception, but none were found.");
+        DeferredExceptions.Count.ShouldBe(1, $"Expected a deferred exception, but {DeferredExceptions.Count} were found: {new AggregateException(DeferredExceptions)}.");
```

Comment:
there's a `ShouldNotBeEmpty` and `ShouldHaveSingleItem` that would better suite here as the error messages will show what values are in the list instead.